replay_log.py

```diff
@@ -99,47 +86,45 @@ def replay_log(log_file, request_to_file_uuid, enable_concurrency = True):
   # uuid of file that's concurrently running
   concurrent_uuid = None
 
+  last_timestamp = 0
+
   for line in fd:
-    request_source, request_content, request_size, concurrent = line.split('\t')
-    concurrent = concurrent.strip() # strip newlines
-    print 'request_source: ' + str(request_source)
-    print 'request_content: ' + str(request_content)
-    print 'request_size: ' + str(request_size)
-    print 'concurrent: ' + str(concurrent)
-    uuid = request_to_file_uuid[request_content]
+    timestamp, uuid, source, source_uuid, dest, request_type, response_code, response_size = line.split('\t')
 
+    if request_type == WRITE_REQUEST and not allow_writes:
+      continue
+      
     # If concurrent, run concurrently with delay
-    if enable_concurrency and concurrent == 'C':
-      concurrent_processes, concurrent_uuid = check_concurrent_execution_and_wait(concurrent_processes, concurrent_uuid, uuid)
+    if enable_concurrency and timestamp == last_timestamp:
+      concurrent_processes = check_concurrent_execution_and_wait(concurrent_processes, uuid)
```

Comment:
Is this correct? `check_concurrent_execution_and_wait` will always wait for all the processes in the queue to finish because of `process.join()` on all the processes and the queue will always be empty returning from the method.

This is a correct implementation only if all the processes are already on the queue, but here it is being executed at each loop.